libbeat/idxmgmt/ilm/client_handler.go

```diff
@@ -205,6 +207,77 @@ func (h *esClientHandler) queryFeatures(to interface{}) (int, error) {
 	return status, nil
 }
 
-func (h *esClientHandler) access() ESClient {
-	return h.client
+// FileClientHandler implements the ClientHandler interface for writing to a file.
+type FileClientHandler struct {
+	client FileClient
+}
+
+// FileClient defines the minimal interface required for the ClientHandler to
+// prepare a policy and write alias.
+type FileClient interface {
+	GetVersion() common.Version
+	Write(name string, body string) error
+}
+
+// NewFileClientHandler initializes and returns a new FileClientHandler instance.
+func NewFileClientHandler(c FileClient) *FileClientHandler {
+	return &FileClientHandler{client: c}
+}
+
+// CheckILMEnabled indicates whether or not ILM is supported for the configured mode and client version.
+func (h *FileClientHandler) CheckILMEnabled(mode Mode) (bool, error) {
+	if mode == ModeDisabled {
+		return false, nil
+	}
+
+	avail, probe := checkILMVersion(mode, h.client.GetVersion())
+	if !avail {
+		if mode == ModeEnabled {
+			ver := h.client.GetVersion()
+			return false, errf(ErrESVersionNotSupported,
+				"Elasticsearch %v does not support ILM", ver.String())
+		}
+		return false, nil
+	}
+
+	if !probe {
+		// version potentially supports ILM, but mode + version indicates that we
+		// want to disable ILM support.
+		return false, nil
+	}
+	return true, nil
```

Comment:
normally I'd say `return probe, nil`. Yet the code comment is helpful. Maybe better place explaining `probe` would be `checkILMVersion` (which is used by ES client as well).